backend/raw_data_processor.py

```diff
@@ -13,37 +13,83 @@
 # =============================================================================
 
 """A Module for processing raw data."""
-
-
+from google.api_core.exceptions import RequestRangeNotSatisfiable
+from google.api_core.exceptions import NotFound
 from utils import parse_csv_line
+SIZE_ONE_LINE = 50
 
 
 class RawDataProcessor:
     """Class for processing raw data."""
 
-    def __init__(self, rawfile, number_per_slice):
-        self._rawfile = rawfile
+    def __init__(self, rawfile, number_per_slice, bucket=None):
+        self._blob = None
+        self._bucket = bucket
+        self._eof = False
+        self._file = None
+        self._file_pointer = 0
+        self._loaded_records = []
         self._number_per_slice = number_per_slice
-        self._file = open(rawfile, 'r')
+        self._rawfile = rawfile
+
+        if bucket is None:
+            self._file = open(rawfile, 'r')
+        else:
+            self._blob = self._bucket.blob(self._rawfile)
 
     def read_next_slice(self):
         """Reads raw data for a single slice.
```

Comment:
Reads next slice of raw data. 
Also explain how u find "next slice"?

Reply:
Updated in class docstring.